### ISSUE TITLE
Wayland not supported by AppImage

### ISSUE THREAD

**Issue by Max-Might** [moonlight-stream/moonlight-qt#916]
**Describe the bug**
Launching the AppImage and forcing it to use wayland environment (gnome on wayland) throws an error.

**Steps to reproduce**
Just executing the AppImage seems to run fine but shows the following message:
```
Detected XWayland. This will probably break hardware decoding! Try running with QT_QPA_PLATFORM=wayland or switch to X11.
```

And it looks like the hardware decoding is not working indeed. So I tried to run the AppImage with the suggested environment variable but it could not find the wayland platform files inside the AppImage:
```
QT_QPA_PLATFORM=wayland ./Moonlight-4.3.1-x86_64.AppImage 
00:00:00 - Qt Info: Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
00:00:00 - Qt Warning: Could not find the Qt platform plugin "wayland" in ""
00:00:00 - Qt Fatal: This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: xcb.

Aborted (core dumped)
```

**Client PC details (please complete the following information)**
 - OS: Arch Linux  December 2022
 - Moonlight Version: 4.3.1
 - GPU: Nvidia 9500M (nouveau driver)

**Additional context**
Looking around it looks like the error is fixed by installing the `qt5-wayland` package (for the specific distribution) and I did it on my client PC but it still fails because the wayland platform files need to be inside the AppImage.

**Comment by cgutman** [comment]
Yeah, this is why the AppImage is marked with "not recommended on Wayland".

The reason we don't do this is because EGL is broken in AppImages with Wayland. I haven't fully debugged it but I think it's a mismatch between AppImage and host Mesa libraries.

If you want Wayland support, use the Flatpak or Snap packages (or AUR, since you're on Arch).

**Comment by dantti** [comment]
Hi, I stumbled open this issue while trying to add Wayland for my apps, so in case you are interested in AppImage + Wayland just use the plugin from my fork (waiting long for a few patches to get upstreamed...), https://github.com/dantti/linuxdeploy-plugin-qt/releases/tag/continuous and set the ENV `EXTRA_PLATFORM_PLUGINS=libqwayland-egl.so;libqwayland-generic.so`

**Comment by Dan-Do** [comment]
Hello,
Does anyone know how to fix this error?
```terminal
00:00:00 - Qt Info: Unable to detect Wayland or X11, so EGLFS will be used by default. Set QT_QPA_PLATFORM to override this.
00:00:00 - Qt Info: Setting display mode by default. Set QT_QPA_EGLFS_ALWAYS_SET_MODE=0 to override this.
00:00:00 - Qt Warning: Could not find the Qt platform plugin "eglfs" in ""
00:00:00 - Qt Fatal: This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
Available platform plugins are: xcb.
Aborted                    (core dumped)
```
I am using Fedora rawhide LXQt with labwc compositor
```terminal
Linux 7.1.0-0.rc2.260507g8ab992f815d6.19.fc45.x86_64 #1 SMP PREEMPT_DYNAMIC Thu May  7 20:08:26 UTC 2026 x86_64 GNU/Linux
```

**Comment by cgutman** [comment]
@Dan-Do Please open a new issue for this. I don't think it's related to this request.
